rewards/src/lib.rs

```diff
@@ -329,11 +329,15 @@ impl<T: Config> Pallet<T> {
 		move_share: T::Share,
 		other: &T::AccountId,
 	) -> DispatchResult {
-		SharesAndWithdrawnRewards::<T>::mutate(pool, other, |increased_share| {
+		SharesAndWithdrawnRewards::<T>::try_mutate(pool, other, |increased_share| {
 			let (increased_share, increased_rewards) = increased_share;
-			SharesAndWithdrawnRewards::<T>::mutate_exists(pool, who, |share| {
+			SharesAndWithdrawnRewards::<T>::try_mutate_exists(pool, who, |share| {
 				let (share, rewards) = share.as_mut().ok_or(Error::<T>::ShareDoesNotExist)?;
 				ensure!(move_share < *share, Error::<T>::CanSplitOnlyLessThanShare);
```

Comment:
should we add like ED for shares? It can be a const with min possible amount so transfer or withdrawal won't leave dust

Reply:
yeah in next PR